src/rd.h

```diff
@@ -446,3 +447,5 @@ typedef struct rd_chariov_s {
         char  *ptr;
         size_t size;
 } rd_chariov_t;
+
+#endif /* NDef RD_H */
```

Comment:
Well I have a problem with confusing ! and NDef iow
#if FOO
#endif /* !FOO */
and
#ifndef FOO
#endif /* NDef FOO */

oh and fyi I tend to use

#if FOO
#else /* !FOO */
#endif /* !FOO */

when there are if /else construct.. iow the endif comment is related to the section it close not the section that opened the #if 

Using ! for def/ndef is ambiguous and confusing. I do not mind an alternative suggestion rather than NDef  but it must be different than the existing ! which already has an assigned meaning.

Now you are the maintainer of this codebase, if you insist on confusing convention, I'll comply, albeit under protest :-)